### PR TITLE
RD-6385 Introducing "Deploy On" dropdown to Create Deployment modal

### PR DESCRIPTION
**Pull request by Vorbert-Kruk**
## Description

Changes consisting of:
- Resolving bug, discovered in [DeploymentIdInputField.tsx](https://github.com/cloudify-cosmo/cloudify-stage/pull/2324/files/45d4756b1c758b1f2167b118762d1e2abf3803fc#diff-61b21bbdba9e2d39a578dccc01de389735524e6b5a37b55dd1cd34e3a2aa30bd)
- Connecting new DynamicDropdown to the form and displaying it depending on the blueprint `requirements` property value
- Providing additional test scenario
- Migrating away from implicit lodash imports

### Before
Before

### After
After

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

 

System tests has been executed for the following test scenarios:
- deployment_button_spec,
- deployments_view_spec,
- blueprints_spec,
- blueprint_actions_spec,

as during my investigation I've discovered that widgets related to those test scenarios are affected by my changes

[Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/4114/)
Failing because of the creating deployment timeout in `deployment_button_spec.ts` (exceeded 30s)   

[Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/4119/)
Failing because of the creating deployment timeout in `deployment_button_spec.ts`   

[Build Status](https://jenkins.cloudify.co/view/UI%20Health%20View/job/Stage-UI-System-Test/4133/)
All failing tests (except one) are the exact same tests that are failing on the nightly job
One, not failing on the nightly, has been introduced via #2321 and will be fixed as a part of #2337
So from my side, changes introduced within this PR are not breaking anything 💪 

## Documentation

Documentation update will take place as a part of RD-6444

### CODE REVIEW

File: widgets/common/src/inputs/fields/DeploymentIdInputField.tsx [hunk 1]
```diff
@@ -7,7 +7,7 @@ export default function DeploymentIdInputField({
     onChange,
     ...restProps
 }: DynamicDropdownInputFieldProps) {
-    const fetchUrl = '/searches/deployments?_include=id,display_name';
```
Comment (Vorbert-Kruk) [inline widgets/common/src/inputs/fields/DeploymentIdInputField.tsx:10]: ### Note
I've discovered that it wasn't possible to gather any data with the usage of this component (we were getting 405 HTTP error)
After consultation the situation with core team mamber, it turned out that we were using URL to the endpoint that have never existed 🤯

Our suspicion was that during input components relocation (which has taken place some time ago) urls were mistakenly unified 

But at least the issue has been spotted and fixed 💪 
Just wanted to mention the possible background of the issue, to make sure that in future CRs we would be more aware about possibilities of introducing similar issues 🎉

**Comment by Vorbert-Kruk** [comment]
Hi @kubama, as @qooban is taking vacation, starting next week and I'm taking a day off tomorrow
I thought that it would be wise to ask you about the CR 💪 

I know that currently there are some ongoing matters (like issues related to `i18n`) so don't worry about this CR when it comes to priority 😁 
As I won't be available tomorrow, we can have a some delay room for the CR :)

**Review by kubama — COMMENTED** [review]
Looks nice and clean, I only have one comment.

File: app/widgets/common/deployModal/GenericDeployModal.utils.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+interface DeployOnDropdownItem {
+    id: string;
+    // eslint-disable-next-line camelcase
+    display_name: string;
+}
+
+export const deployOnTextFormatter = (item: DeployOnDropdownItem) => {
+    return Stage.Utils.formatDisplayName({ id: item.id, displayName: item.display_name });
```
[the remark this answers is not in the document]
Comment (Vorbert-Kruk) [reply app/widgets/common/deployModal/GenericDeployModal.utils.ts]: Sure
Applied as a part of 1a44103ca89590e03b61cd80ebbec92a493dcf39 🎉

**Review by kubama — APPROVED** [review]

**Pull request merged**